datasafe-business-tests/datasafe-business-tests-random-actions/src/test/java/de/adorsys/datasafe/business/impl/e2e/randomactions/RandomActionsOnSimpleDatasafeAdapterTest.java

```diff
@@ -0,0 +1,225 @@
+package de.adorsys.datasafe.business.impl.e2e.randomactions;
+
+import de.adorsys.datasafe.business.impl.e2e.randomactions.framework.BaseRandomActions;
+import de.adorsys.datasafe.business.impl.e2e.randomactions.framework.services.StatisticService;
+import de.adorsys.datasafe.business.impl.service.DefaultDatasafeServices;
+import de.adorsys.datasafe.directory.api.profile.operations.ProfileOperations;
+import de.adorsys.datasafe.directory.api.types.CreateUserPrivateProfile;
+import de.adorsys.datasafe.directory.api.types.CreateUserPublicProfile;
+import de.adorsys.datasafe.directory.api.types.UserPrivateProfile;
+import de.adorsys.datasafe.directory.api.types.UserPublicProfile;
+import de.adorsys.datasafe.encrypiton.api.types.UserID;
+import de.adorsys.datasafe.encrypiton.api.types.UserIDAuth;
+import de.adorsys.datasafe.inbox.api.InboxService;
+import de.adorsys.datasafe.privatestore.api.PrivateSpaceService;
+import de.adorsys.datasafe.simple.adapter.api.SimpleDatasafeService;
+import de.adorsys.datasafe.simple.adapter.api.exceptions.SimpleAdapterException;
+import de.adorsys.datasafe.simple.adapter.api.types.*;
+import de.adorsys.datasafe.simple.adapter.impl.SimpleDatasafeServiceImpl;
+import de.adorsys.datasafe.types.api.actions.ListRequest;
+import de.adorsys.datasafe.types.api.actions.ReadRequest;
+import de.adorsys.datasafe.types.api.actions.RemoveRequest;
+import de.adorsys.datasafe.types.api.actions.WriteRequest;
+import de.adorsys.datasafe.types.api.resource.*;
+import lombok.RequiredArgsConstructor;
+import org.junit.jupiter.api.condition.DisabledIfSystemProperty;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.*;
+import java.nio.file.FileSystems;
+import java.time.Instant;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static de.adorsys.datasafe.business.impl.e2e.randomactions.framework.BaseRandomActions.DISABLE_RANDOM_ACTIONS_TEST;
+
+/**
+ * Executes random user actions in multiple threads against Datasafe-core.
+ * We have action fixture for 10 users, where each user does share,read,write,etc. After one executes
+ * actions in this fixture he can validate inbox and private directory content using fixture expectation section.
+ * This fixture is duplicated N times and submitted to thread pool, so any thread in pool can pick some action and
+ * act independently of others. Thread actions and expectations are prefixed with execution id.
+ * Imitates close-to-production SimpleDatasafeAdapter deployment.
+ */
+@DisabledIfSystemProperty(named = DISABLE_RANDOM_ACTIONS_TEST, matches = "true")
```

Comment:
The whole build in travis for doc 208 now takes 4 times  longer (e.g. 1 hour rather than 15 mins) than before. Will this become the default? Why not a property that has to be switched on?
(btw. is this not the same like my SKIP_CEPH option, I wanted to have but was not allowed to put in? - currently I can no more work on my macpro, as all ceph tests fail. So I test my modules and compile everything else with -Dskiptests. I would be very happy to switch of ceph and ceph tests with a system property)

Reply:
Yes it will be new default, i will check possible optimization (maybe we can drop concurrency tests for FS since it is not production setup and keep only minio)
This test suite is additional - not core one (we do already have concurrency tests in core suite), it resides in completely separate package and is a member of durability test class, this is why it can be disabled using DISABLE_RANDOM_ACTIONS_TEST

Regarding CEPH - you can increase container start timeout
```
cephContainer = new GenericContainer("ceph/daemon")
                .withExposedPorts(8000)
                .withEnv("RGW_FRONTEND_PORT", "8000")
                .withEnv("SREE_PORT", "5000")
                .withEnv("DEBUG", "verbose")
                .withEnv("CEPH_DEMO_UID", "nano")
                .withEnv("MON_IP", "127.0.0.1")
                .withEnv("CEPH_PUBLIC_NETWORK", "0.0.0.0/0")
                .withEnv("CEPH_DAEMON", "demo")
                .withEnv("DEMO_DAEMONS", "mon,mgr,osd,rgw")
                .withEnv("CEPH_DEMO_ACCESS_KEY", cephAccessKeyID)
                .withEnv("CEPH_DEMO_SECRET_KEY", cephSecretAccessKey)
                .withCommand("mkdir -p /etc/ceph && mkdir -p /var/lib/ceph && /entrypoint.sh")
                .waitingFor(Wait.defaultWaitStrategy().withStartupTimeout(Duration.ofSeconds(60)));
```

Reply:
> currently I can no more work on my macpro, as all ceph tests fail.
Additionally to increasing test timeout i can suggest doing
`mvn clean verify -fn`
This will not fail build on first test failures - it will run all, from final output you can simply validate that only CEPH caused problems

Reply:
thanks, I will try this. 
Maybe you add different profiles to make it easer to skip the tests locally?